design-documents/0003-gene-sets/text.md

```diff
@@ -0,0 +1,263 @@
+
+- gene_set_name
+- gene_set_description
+- gene_symbol
+- gene_description
+
+**gene_set_name** and **gene_symbol** must contain a string value - ie, must not be empty or null. The description columns may be empty, but are also string values. The file may contain additional columns, which will be stored as a dict of additional params {“column_name”: “value”} on the gene.
+
+Where a geneset has multiple gene rows, and they differ in gene_set_description value, the description in the first row of the geneset will be used. For further notes on format and validation rules see [gene sets formats](https://github.com/brianraymor/single-cell-curation/blob/gene-sets-format/docs/gene_sets.md)
+
+```csv
+geneset_name,geneset_description,gene_symbol,gene_description,provenance
+club.cell,Co-expressed in club cells,CCKAR,This gene encodes a G protein,Pubmed ID XYZ123
+club.cell,Co-expressed in club cells,SCGB3A2,"Coding, assoc. with asthma",Pubmed ID ABC456
+club.cell,Co-expressed in club cells,CYP2F2,Protein coding gene associated with metabolic processes,Pubmed ID DCF678
+Macrophage,Co-expressed for macrophages,CD68,Critical to macrophages
+macrophage,Co-expressed for macrophages,CD163,Also critical to macrophages
+```
+
+If additional fields are included, when present for a gene they will be stored as a dict of {header: value} .
+
+### Genesets JSON
+
+The genesets default JSON data structure, used by the portal front-end to save new genesets.
+
+```json
+{
+  "genesets": [
+    {
+      "gene_set_name": "<string>",          # gene set name
+      "gene_set_description": "<string>",   # gene set description/notes
+      "genes": [                            # genes in the gene set
+        {
+          "gene_symbol": "<string>",        # gene identifier/name
+          "gene_description": "<string>"    # description/note for this gene
+          "additional_params" : {      # additional parameters in csv
+  "Header": "value"
+  }
+        },
+        ...                                 # additional genes
+      ]
+    },
+    ...                                     # additional geneset
+  ],
+}
+```
+
+Geneset and gene order is significant, the cellxgene UI will attempt to render in this order. The data portal must preserve this ordering. **geneset_name** must be unique to the entire collection, and each **gene_symbol** in a geneset must be unique (ie, occur only once in a given geneset).
+
+### Abbreviated Genesets JSON
+
+The gene set and dataset tabs of the data portal collections page do not require gene level information so that will not be included with geneset information for a collection.
+
+```json
+[
+  {
+     "id": "<string>",           # gene set id
+     "name": "<string>",         # gene set name
+     "description": "<string>",  # gene set description/notes
+     "linked_datasets": [        # list of ids of datasets linked to geneset
+        datasest_id,
+        ...
+      ]
+  },
+  ...        # additional geneset
```

Comment:
Do you know where these weird red highlightings are coming from?

Reply:
I think its because I specified the code block was json but comments arent actually allowed in json. I want to leave the comments so I'll just change the code block type to text